src/lib/ui-element.coffee

```diff
@@ -0,0 +1,18 @@
+class UIElement
```

Comment:
I'm not sure about the logic. I was thinking maybe we'd have a debug flag in the game config and could log drawing/updating etc in this base class.
The main draw for me was to be like "these are the functions a class should implement" so the error you suggest could help enforce this.